### PR TITLE
[IMP] developer: add sections on pyjs and domains

### PR DESCRIPTION
**Pull request by ged-odoo**

### CODE REVIEW

File: content/developer/reference/javascript/framework_overview.rst [hunk 1]
```diff
@@ -307,8 +307,182 @@ and its context will be extended with the `default_period_id` value. This is a
...
+  :returns: any the resulting value of the expression, with respect to the context
+
 .. _javascript/bus:
 
+Domains
+=======
+
+Broadly speaking, domains in Odoo represent a set of records that matches some
+specified conditions. In javascript, they are usually represented either as a
+list of conditions (or of operators: `|` or `&` in prefix notation), or as string
```
Comment (aab-odoo) [inline content/developer/reference/javascript/framework_overview.rst:365]: also operator !

**Review by aab-odoo — APPROVED** [review]

**Comment by ged-odoo** [comment]
another brick in the wall... @odoo/doc-review

**Comment by ged-odoo** [comment]
just rebased

File: content/developer/reference/frontend/framework_overview.rst [hunk 2]
```diff
@@ -310,8 +310,182 @@ and its context will be extended with the `default_period_id` value. This is a
...
+  :param Object context: an object that provides an additional evaluation context
+  :returns: any the resulting value of the expression, with respect to the context
+
+.. js:function:: evaluateExpr(expr[, context])
+
+  :param string expr: a string representing a valid python expression
+  :param Object context: an object that provides an additional evaluation context
+  :returns: any the resulting value of the expression, with respect to the context
+
 .. _javascript/bus:
```
Comment (AntoineVDV) [inline content/developer/reference/frontend/framework_overview.rst:361]:
```suggestion
.. _javascript/domains:
```

File: content/developer/reference/frontend/framework_overview.rst [hunk 3]
```diff
@@ -310,8 +310,182 @@ and its context will be extended with the `default_period_id` value. This is a
...
+  :param operator: an operator
+  :type operator: 'AND' or 'OR'
+
+  :returns: Domain
+
+  Returns a domain representing either the intersection or the union of all the
+  domains, depending on the value of the operator argument.
+
 Bus
 ===
```
Comment (AntoineVDV) [inline content/developer/reference/frontend/framework_overview.rst:490]:
```suggestion
.. _javascript/bus:

Bus
===
```

**Review by AntoineVDV — COMMENTED** [review]
@robodoo delegate+

**Comment by ged-odoo** [comment]
@robodoo r+

**Pull request closed without merging**